book/src/dev/rfcs/0010-v5-transaction.md

```diff
@@ -0,0 +1,211 @@
+
+The difference between V4 and V5 shielded data is in the spends and outputs. For this reason we need to add V4 and V5 spend and output versions to our code:
+
+```
+pub struct SaplingV4ShieldedData {
+    pub first: Either<SpendV4, OutputV4>,
+    pub rest_spends: Vec<SpendV4>,
+    pub rest_outputs: Vec<OutputV4>,
+    pub binding_sig: Signature<Binding>,
+}
+```
+
+Proposed `SpendV4` and `OutputV4` is now described, this is the same as the current implementation but just changing the struct names. 
+Sapling spend code is located at `zebra-chain/src/sapling/spend.rs`:
+
+```
+pub struct SpendV4 {
+    pub cv: commitment::ValueCommitment,
+    pub anchor: tree::Root,
+    pub nullifier: note::Nullifier,
+    pub rk: redjubjub::VerificationKeyBytes<SpendAuth>,
+    pub zkproof: Groth16Proof,
+    pub spend_auth_sig: redjubjub::Signature<SpendAuth>,
+}
+```
+
+The output code is located at `zebra-chain/src/sapling/output.rs`:
+
+```
+pub struct OutputV4 {
+    pub cv: commitment::ValueCommitment,
+    pub cm_u: jubjub::Fq,
+    pub ephemeral_key: keys::EphemeralPublicKey,
+    pub enc_ciphertext: note::EncryptedNote,
+    pub out_ciphertext: note::WrappedNoteKey,
+    pub zkproof: Groth16Proof,
+}
+```
+
+Now lets see how the v5 transaction is specified in the protocol, this is the second table of [Transaction Encoding and Consensus](https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus).
+
+We propose the following representation for transaction V5 in Zebra:
+
+```
+V5 {
+    lock_time: LockTime,
+    expiry_height: block::Height,
+    inputs: Vec<transparent::Input>,
+    outputs: Vec<transparent::Output>,
+    shielded_data: Option<SaplingV5ShieldedData>,
+    value_balance_sapling: Amount,
+    output_proofs_sapling: Vec<Groth16Proof>
+    anchor_sapling: tree::Root,
+    spend_zkproof_sapling: Vec<Groth16Proof>,
+    spend_auth_sigs_sapling: Vec<redjubjub::Signature<SpendAuth>>
+    orchard_data: Option<OrchardData>,
+    flags_orchard: u8,
+    value_balance_orchard: Amount,
+    anchor_orchard: tree::Root,
+}
+```
+
+`SaplingV5ShieldedData` is different from `SaplingV4ShieldedData` so a new type will be defined and implemented in `zebra-chain/src/transaction/shielded_data.rs`.
+Definition will look as follows:
+
+```
+pub struct SaplingV5ShieldedData {
+    pub first: Either<SpendV5, OutputV5>,
+    pub rest_spends: Vec<SpendV5>,
+    pub rest_outputs: Vec<OutputV5>,
+    pub binding_sig: Signature<Binding>,
+}
+```
+
+This leads to new sapling output and spend types.
+
+V5 sapling spend will be defined and implemented in `zebra-chain/src/sapling/spend.rs`(this is in the same file as `SpendV4`):
+```
+pub struct SpendV5 {
+    pub cv: commitment::ValueCommitment,
+    pub nullifier: note::Nullifier,
+    pub rk: redjubjub::VerificationKeyBytes<SpendAuth>,
+}
```

Comment:
Moved from `V5`, because there is one `zkproof` and one `spend_auth_sig` for each `SpendV5`:
```suggestion
    pub zkproof: Groth16Proof,
    pub spend_auth_sig: redjubjub::Signature<SpendAuth>,
}
```